packages/astro/src/content/types-generator.ts

```diff
@@ -382,7 +382,7 @@ async function writeContentFiles({
 	let contentTypesStr = '';
 	let dataTypesStr = '';
 
-	const collectionSchemasDir = new URL('./collections/', contentPaths.cacheDir);
+	const collectionSchemasDir = new URL('./collections/', settings.dotAstroDir);
```

Comment:
All codegen related features need to create files in `.astro`. The path is not content specific so I extracted it to settings to be able to reuse it across content, actions and env. It will also be helpful when I get to work on https://github.com/withastro/roadmap/discussions/887